add config error in job_status error types (#6363)
braze-inc/braze-docs: _docs/_api/endpoints/cdi/get_job_sync_status.md

| job_status | Explanation |
| --- | --- |

> | running | The job is current running |
> | success | All rows synced successfully |
> | partial | Some rows failed to sync due to errors |
> | error | No rows were synced |
> | config_error | Error in integration configuration, please check your integration setup |

Reviewer: ```suggestion
| `running` | The job is currently running. |
| `success` | All rows synced successfully. |
| `partial` | Some rows failed to sync due to errors. |
| `error` | No rows were synced. |
| `config_error` | There was an error in integration configuration. Check your integration setup. |
```
